connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectStreamHandler.scala

```diff
@@ -174,12 +187,14 @@ object SparkConnectStreamHandler {
 
     dataframe.queryExecution.executedPlan match {
       case LocalTableScanExec(_, rows) =>
+        events.postFinished()
         rowToArrowConverter(rows.iterator).foreach { case (bytes, count) =>
           sendBatch(bytes, count)
         }
       case _ =>
         SQLExecution.withNewExecutionId(dataframe.queryExecution, Some("collectArrow")) {
           val rows = dataframe.queryExecution.executedPlan.execute()
+          events.postFinished()
```

Comment:
The query didn't even start running at this point, so I don't think this should be finished.
The query is only submitted for execution by the `spark.sparkContext.submitJob` further down.
The query is finished when that future.onComplete happens.
In the meantime, the query asynchronously sends results back as soon as they are available.

Reply:
Moved postFinished on onComplete